source/Octopus.Client/Repositories/Async/BasicRepository.cs

```diff
@@ -48,15 +48,24 @@ protected virtual void CheckSpaceResource(IHaveSpaceResource spaceResource)
                 whenSystemScoped: () => { },
                 whenUnspecifiedScope: () =>
                 {
-                    var spaceRoot = Repository.LoadSpaceRootDocument();
+                    var spaceRoot = Repository.LoadSpaceRootDocument().Result;
```

Comment:
We can't use `.Result` here. While this would work in applications like ours, this could introduced deadlocks in other applications that use a different `SynchronizationContext` (eg winforms or .net framework asp.net). The correct way to do this is `var spaceRoot = await Repository.LoadSpaceRootDocument().ConfigureAwait(false);`.

The `.ConfigureAwait(false)` part is also really important for the same reason (prevents deadlocks). This is a really common pattern in async code like this that exists in libraries (if you search for this in this repo, you should find it everywhere we use `await`), but is something that we would tend to omit from code that we don't publish as libraries (eg in our OctopusServer repo).

In fact, this highlights an existing bug - the current implementation doesn't actually work. The line below where it goes `spaceRoot != null` is actually check whether the `Task` is null (which would never happen), rather than the `Task`s result.

The problem here is that you need to keep making everything `async` all the way up to get this to work, so this change will affect at least a few more methods, but I don't think its too much more work in this case.

Let me know if you need a hand with this.